doc/rst/developer/bestPractices/ContributorInfo.rst

```diff
@@ -9,80 +9,297 @@ The git repository for the project is hosted on Github at
 
 See the `Developer Workflow`_ instructions below for setting up a github
 account, signing the contributor agreement, and submitting pull requests.
+This document contains a mixture of tips for git beginners and specific
+Chapel workflow recommendations.
 
 Developer Workflow
 ------------------
 
 Overview:
 
-#. `Fork repo on github`_
-#. `Create new branch`_
-#. `Submit pull request`_
+#. `Discuss design`_ changes or big development efforts
+#. `Set up a branch for development`_
+
+   #. `Set up repository`_
+   #. `Create new branch`_
+
+#. `Develop and test contributions locally`_
+
+   #. `Test your feature`_
+
+#. `Request feedback on your changes`_
+
+   #. `Push your work`_
+   #. `Ask for feedback early`_ (optional)
+   #. `Submit pull request`_
+   #. `Find a reviewer`_
+   #. `Work with your reviewers`_
+
 #. `Test changes`_ (can be done by team at Cray)
-#. `Mail chapel-developers@ for review and merge`_
+#. `Merge changes in to master`_
+#. `Watch automatic testing`_ and address issues
+
+`HOWTO/Git/GitHub details`_
+
+`Policy details`_
 
-.. _Fork repo on github:
+.. _Discuss design:
 
-Fork repo on github
-~~~~~~~~~~~~~~~~~~~
+Discuss design changes or big development efforts
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-These are expected to evolve over time as the governance of Chapel is migrated
-from Cray to an external/community body (the major elements are likely to be
-similar, though the specific people involved are likely to change and grow).
+Before performing major work on the language, please open an issue on the git
+repository and/or send an email to chapel-developers_ with a prefix of
+``[Design]`` in the subject header for discussion with the community and to
+ensure that you are aware of any parallel efforts in that area.
 
-* `Set up a github account`_. The "Free" plan is sufficient for contributing to
-  Chapel.
+.. _Set up a branch for development:
 
-* Make sure you have configured your environment to work with git. See
-  :ref:`Initial Git Set up <initial_git_setup>` instructions.
+Set up a branch for development
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-* If you will need commit/push access to the main repository,
-  `chapel-lang/chapel`_, send a request including your github username to
-  chapel_admin _at_ cray.com.
+This consists of two sets of operations.  The first (`Set up repository`_) only
+needs to happen once per developer.  The second (`Create new branch`_) should
+happen for every new effort.
 
-  .. note::
+.. _Set up repository:
 
-    You do not need commit/push access to the main repo in order to contribute
-    code. Reviewers on the core team can pull, review, and merge your pull
-    requests.
+Set up repository
```

Comment:
Okay, I'm having a bit of trouble with this one, so I figured I should ask.  I see what you mean about the first three bullets, but they are already mostly just links and I think the remainder should be at that location still.  I moved a couple of the bullets around, since "who has commit access" and "third party code" weren't actually steps.  After the modifications that have happened, what would you prefer I do?